Sources/PaltaLib/Engine/PaltaLib.swift

```diff
@@ -41,7 +78,135 @@ public final class PaltaLib {
         if let serverURL = target.serverURL {
             amplitudeInstance.setServerUrl(serverURL.absoluteString)
         }
-
         amplitudeInstances.append(amplitudeInstance)
     }
+    
+    public func initializeApiKey(apiKey: String) {
+        amplitudeInstances.forEach {
+            $0.initializeApiKey(apiKey)
+        }
+    }
+    
+    public func initializeApiKey(apiKey: String, userId: String?) {
+        amplitudeInstances.forEach {
+            $0.initializeApiKey(apiKey, userId: userId)
+        }
+    }
+    
+//    public func setOptOut(_ enabled: Bool) {
+//        amplitudeInstances.forEach {
+//            $0.setOptOut(enabled)
+//        }
+//    }
+    
+    public func setOffline(_ offline: Bool) {
+        amplitudeInstances.forEach {
+            $0.setOffline(offline)
+        }
+    }
+    
+    public func useAdvertisingIdForDeviceId() {
+        amplitudeInstances.forEach {
+            $0.useAdvertisingIdForDeviceId()
+        }
+    }
+
+    public func setTrackingOptions(_ options: AMPTrackingOptions) {
+        amplitudeInstances.forEach {
+            $0.setTrackingOptions(options)
+        }
+    }
+
+    public func enableCoppaControl() {
+        amplitudeInstances.forEach {
+            $0.enableCoppaControl()
+        }
+    }
+    
+    public func disableCoppaControl() {
+        amplitudeInstances.forEach {
+            $0.disableCoppaControl()
+        }
+    }
+    
+    public func setServerUrl(_ serverUrl: String) {
+        amplitudeInstances.forEach {
+            $0.setServerUrl(serverUrl)
+        }
+    }
+
+    public func setContentTypeHeader(_ contentType: String) {
+        amplitudeInstances.forEach {
+            $0.setContentTypeHeader(contentType)
+        }
+    }
+    
+    public func setBearerToken(_ token: String) {
+        amplitudeInstances.forEach {
+            $0.setBearerToken(token)
+        }
+    }
+    
+    public func setPlan(_ plan: AMPPlan) {
+        amplitudeInstances.forEach {
+            $0.setPlan(plan)
+        }
+    }
+
+    public func setServerZone(_ serverZone: AMPServerZone) {
```

Comment:
This one and the one below are only relevant for amplitude as it update endpoints to be used by the sdk